source/math/FloatBox2D.ooc

```diff
@@ -87,18 +87,10 @@ FloatBox2D: cover {
 		This new(left, top, width, height)
 	}
 	union: func ~box (other: This) -> This { // Rock bug: Union without suffix cannot be used because the C name conflicts with keyword "union"
-		left := Float minimum(this left, other left)
-		top := Float minimum(this top, other top)
-		width := Float maximum(0, (Float maximum(this right, other right) - left))
-		height := Float maximum(0, (Float maximum(this bottom, other bottom) - top))
-		This new(left, top, width, height)
+		This new(this leftTop minimum(other leftTop), this rightBottom maximum(other rightBottom))
```

Comment:
The discussion in #783 seems to show that size *should be* nonnegative. And you're right, this seems to work just fine now.